ext/openssl/ossl_ssl.c

```diff
@@ -1644,9 +1644,11 @@ ossl_ssl_initialize(int argc, VALUE *argv, VALUE self)
     RTYPEDDATA_DATA(self) = ssl;
 
     SSL_set_ex_data(ssl, ossl_ssl_ex_ptr_idx, (void *)self);
+    RB_OBJ_WRITTEN(self, Qundef, self);
```

Comment:
Yes, you are right, these two `RB_OBJ_WRITTEN` aren't really necessary, I added them mostly for consistency.

> I also think that we don't need ossl_sslctx_mark() and  rb_gc_mark((VALUE)SSL_get_ex_data(ssl, ossl_ssl_ex_ptr_idx))

Hum, I think we may need it, not because we need to mark the object, but because we the object need to be pined, and `rb_gc_mark` does that.

Reply:
Ah, I see.

Reply:
> Hum, I think we may need it, not because we need to mark the object, but because we the object need to be pined, and rb_gc_mark does that.

@byroot I assumed that your final conclusion was "we may need it (= `RB_OBJ_WRITTEN()` after `SSL_CTX_set_ex_data()`)". But I don't see it in this PR. What's the reason of the decision?

Reply:
No, here we don't need it. Write Barrier are needed to notify the GC that a reference was created between two objects. Here we assign `self`, so it's not really a reference.

I knew that initially but added it anyway for consistency.

But now that it appear they were all useless, I figured we might as well remove them all.

Reply:
Okay. Thanks for the explanation. Now I can see you only added the `RB_OBJ_WRITTEN()` to the necessary parts.